test/__snapshots__/StatsTestCases.test.js.snap

```diff
@@ -1130,20 +1130,18 @@ Child
     Built at: Thu Jan 01 1970 00:00:00 GMT
                                      Asset       Size        Chunks             Chunk Names
         a-all~main-0034bb84916bcade4cc7.js  154 bytes      all~main  [emitted]  all~main
-            a-main-9407860001b0bf9acb00.js  108 bytes          main  [emitted]  main
```

Comment:
What was it that 108 bytes for? Bootstrap code?

Reply:
This piece of code is actually pretty important.

`[[0,"runtime~main","all~main"]]` means `execute module 0 when chunks "runtime~main","all~main" are loaded`, so it can't be removed.

You need to check if a Chunk contains any entryModule. -> `ChunkGraph.getNumberOfEntryModules(chunk)`

It's also possible that a Chunk contains prefetch specification -> `chunk.getChildIdsByOrders(chunkGraph)`.

But since this is enviroment-specific, it should be implemented in JsonpChunkTemplate. Therefore it's required to add a hook (maybe `shouldRenderChunk`) to the JavascriptModulesPlugin to allow adding this logic.

Reply:
kk, I used the `ChunkGraph.getNumberOfEntryModules(chunk)` already for a pseudo test

**`webpack.config.js`**
```js
const config = {
   entry: {
     remove: './empty.js'
   }
}
```

&&

```
import(`./empty.js`).then((module) => module)
```

which initially leaded me to use `module.size() > 0`  in `hasModules`, but yep the chunk is still connected/referenced and rendered in `JsonpMainTemplatePlugin`. The `RemoveEmptyChunksPlugin` also disconnects and deletes the chunk(s), so I kind of wondered about that... :)

I will take a look and try...

Reply:
For a testcase you could disable `optimization.removeEmptyChunks`. In reality this can happend with the `mini-extract-css-plugin` when CSS is separated from JS. In this case chunks are created which contain only CSS.